utils/ixdtf/src/parsers/mod.rs

```diff
@@ -62,22 +62,82 @@ impl<'a> IxdtfParser<'a> {
     ///
     /// [temporal-dt]: https://tc39.es/proposal-temporal/#prod-TemporalDateTimeString
     pub fn parse(&mut self) -> ParserResult<IxdtfParseRecord<'a>> {
-        datetime::parse_annotated_date_time(&mut self.cursor)
+        self.parse_with_annotation_handler(|annotation| {
+            if annotation.critical && annotation.key != "u-ca" {
+                Err(ParserError::UnrecognizedCritical)
+            } else {
+                Ok(())
+            }
+        })
+    }
+
+    // TODO: Build out docs after consensus
+    /// Parses the source as an annotated DateTime string with an Annotation handler.
+    pub fn parse_with_annotation_handler(
+        &mut self,
+        handler: impl FnMut(Annotation<'a>) -> ParserResult<()>,
+    ) -> ParserResult<IxdtfParseRecord<'a>> {
+        datetime::parse_annotated_date_time(&mut self.cursor, handler)
     }
 
     /// Parses the source as an annotated YearMonth string.
     pub fn parse_year_month(&mut self) -> ParserResult<IxdtfParseRecord<'a>> {
-        datetime::parse_annotated_year_month(&mut self.cursor)
+        self.parse_year_month_with_annotation_handler(|annotation| {
+            if annotation.critical && annotation.key != "u-ca" {
+                Err(ParserError::UnrecognizedCritical)
+            } else {
+                Ok(())
+            }
+        })
+    }
+
+    // TODO: Build out docs after consensus
+    /// Parses the source as an annotated YearMonth string with an Annotation handler.
+    pub fn parse_year_month_with_annotation_handler(
+        &mut self,
+        handler: impl FnMut(Annotation<'a>) -> ParserResult<()>,
+    ) -> ParserResult<IxdtfParseRecord<'a>> {
+        datetime::parse_annotated_year_month(&mut self.cursor, handler)
     }
 
     /// Parses the source as an annotated MonthDay string.
     pub fn parse_month_day(&mut self) -> ParserResult<IxdtfParseRecord<'a>> {
-        datetime::parse_annotated_month_day(&mut self.cursor)
+        self.parse_month_day_with_annotation_handler(|annotation| {
+            if annotation.critical && annotation.key != "u-ca" {
+                Err(ParserError::UnrecognizedCritical)
+            } else {
+                Ok(())
+            }
+        })
```

Comment:
Thought: Maybe the signature should be `(Annotation) -> Option<Annotation>` where it returns back the annotation if it didn't understand the annotation. Then the critical logic can be handled where it is supposed to be handled and there are fewer invariants for the user to enforce.

```suggestion
        self.parse_month_day_with_annotation_handler(|annotation| Some(annotation))
```